types/merge-source-map.d.ts

```diff
@@ -1,17 +1,28 @@
-
+/**
+ * Ambient module declaration for the [merge-source-map library](https://github.com/keik/merge-source-map).
+ *
+ * At the time of this writing, no official nor third party type declaration files exist for v1.1.0 of library, and are
+ * recreated here for type safety purposes.
+ */
 declare module 'merge-source-map' {
-  interface SourceMap {
```

Comment:
There are so many representations of SourceMaps in the codebase. One from:
- terser
- MagicString
- Stencil
- rollup

This is an attempt to coalesce them, because honestly it gets confusing really quickly which definition you're dealing with.  I don't want to combine them all into one definition, but this particular definition seemed extraneous